src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -983,240 +826,66 @@ public SolutionState WithProjectMetadataReferences(ProjectId projectId, IReadOnl
         /// Create a new solution instance with the project specified updated to include the
         /// specified analyzer references.
         /// </summary>
-        public SolutionState AddAnalyzerReferences(ProjectId projectId, ImmutableArray<AnalyzerReference> analyzerReferences)
+        public (SolutionState, ProjectState oldProject, ProjectState newProject) AddAnalyzerReferences(ProjectId projectId, ImmutableArray<AnalyzerReference> analyzerReferences)
```

Comment:
note: this method returns the new solution, and the old/new versions of the project that were updated.  this is needed by the caller as `oldProject` is used in the translation action.

Note: it is only used to determine the project language.  So it's likely we can simplify this to just return the new project as the language should be the same.